### ISSUE TITLE
Performance regression in <Copy/> task with dev16

### ISSUE THREAD

**Issue by jonathanpeppers** [Microsoft/msbuild#4065]
### Steps to reproduce

Checkout my example repro here (`dev16-repro` branch): https://github.com/jonathanpeppers/HelloWorld/tree/dev16-repro

I have the Xamarin workload installed and an Android device connected via USB. An emulator would also be fine.

Command line
```
.\build.ps1
```
This script is running Xamarin.Android's `Install` target to deploy an APK. It is timing an initial deploy, then a deploy with no changes.

Reviewing `/clp:peformancesummary`:
```
# 16.0
434 ms  _CopyFilesMarkedCopyLocal                  1 calls
# 15.9
 20 ms  _CopyFilesMarkedCopyLocal                  1 calls
```
The bulk of the time here is spent in `<Copy/>`:
```
# 16.0
433 ms  Copy                                       5 calls
# 15.9
23 ms  Copy                                       5 calls
```
These two files are ~6k and ~1k...

![image](https://user-images.githubusercontent.com/840039/51045918-6115ce80-158a-11e9-85b6-74f7e9b04290.png)

What is weird about this, is I couldn't reproduce this by just running the `Build` target.

### Expected  behavior

`<Copy/>` should be quick when no changes are occurring.

### Actual behavior

In some cases, `<Copy/>` is taking a bit of time, even when doing no work against two small files.

### Environment data
`msbuild /version` output:
```
Microsoft (R) Build Engine version 16.0.360-preview+g9781d96883 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

16.0.360.33175
```
OS info:
```
Microsoft Windows 10 Enterprise
10.0.17763 Build 17763
```

Binlogs: [logs.zip](https://github.com/Microsoft/msbuild/files/2750361/logs.zip)

**Comment by danmosemsft** [comment]
There were a series of issues in .NET Core related to the nanosecond parts of file timestamps - either not reading them correctly, or when doing `File.Copy` not copying them fully. These should all be corrected now in CoreFX master. The effect of this was that MSBuild could see the destination as older even when it wasn't.

You can verify this was the problem by using `touch` to set the timestamp of source and destination files to something (say a whole number of seconds). Repeat the build and it should not copy.

https://github.com/dotnet/corefx/issues/32115
https://github.com/dotnet/corefx/issues/31379
https://github.com/dotnet/corefx/issues/26024

**Comment by jonathanpeppers** [comment]
@danmosemsft I'm not sure these are the issue. These builds are running on MSBuild full .NET framework on Windows.

Reviewing the binlog, the files aren't copying, but the task takes 426ms:
```
Copy
    Assembly = Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, 
...
    Did not copy from file "C:\Users\jopepper\Desktop\Git\HelloWorld\HelloForms\HelloForms\bin\Debug\netstandard2.0\HelloForms.dll" to file "bin\Debug\HelloForms.dll" because the "SkipUnchangedFiles" parameter was set to "true" in the project and the files' sizes and timestamps match.
    Did not copy from file "C:\Users\jopepper\Desktop\Git\HelloWorld\HelloForms\HelloForms\bin\Debug\netstandard2.0\HelloForms.pdb" to file "bin\Debug\HelloForms.pdb" because the "SkipUnchangedFiles" parameter was set to "true" in the project and the files' sizes and timestamps match.
```
I believe this is the line these are logged from: https://github.com/Microsoft/msbuild/blob/9781d968839ded744e18152f991d21eb6ce83862/src/Tasks/Copy.cs#L669-L676

**Comment by danmosemsft** [comment]
Ah yes that's not the issue then. I wonder whether procmon might suggest why the time stamp checks are taking so long (eg msybe they are getting repeated? Seems unlikely)

**Comment by BenVillalobos** [comment]
@jonathanpeppers I can't seem to repro your issue with the versions of MSBuild on my machine.

### Versions I tested
```
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
15.9.21.664

-----

Microsoft (R) Build Engine version 16.1.68-preview+g64a5b6be6d for .NET Framework
16.1.68.42340
```

### My Results

```
# 15.9
5 ms  _CopyFilesMarkedCopyLocal                  1 calls
47 ms  Copy                                       8 calls

# 16.1
5 ms  _CopyFilesMarkedCopyLocal                  1 calls
12 ms  Copy                                       5 calls
```

Are you still seeing this regression?

**Comment by jonathanpeppers** [comment]
I can repro with 16.0 GA:

![image](https://user-images.githubusercontent.com/840039/57107764-7f006e00-6cf6-11e9-9b25-dd3bc12a0978.png)

But I think it's fixed with `16.1.68-preview+g64a5b6be6d`:

![image](https://user-images.githubusercontent.com/840039/57107863-cdae0800-6cf6-11e9-8c89-5683173e5ad9.png)

So must be already fixed? I think we can close this, I'll reopen if I see it again with 16.1 or higher.
